### PR TITLE
Migrate TriggerArtifactDefinitionGenerate to prefect

### PR DESCRIPTION
**Pull request by LucasG0**

### CODE REVIEW

File: backend/infrahub/events/tasks.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from prefect import flow
```
Comment (dgarros) [inline backend/infrahub/events/tasks.py]: I would suggest to move this file, either under `git/tasks.py` or to create a new directory named `artifacts`
I was planning to reserve event the definition of the events themselves and to have them separated from the workflows

Comment (LucasG0) [reply backend/infrahub/events/tasks.py]: Thanks, I was not sure where to locate it and put it here by waiting to have more information about it

File: backend/infrahub/git/tasks.py [hunk 2]
```diff
@@ -108,3 +109,16 @@ async def git_branch_create(
     repo = await InfrahubRepository.init(id=repository_id, name=repository_name, client=client)
     async with lock.registry.get(name=repository_name, namespace="repository"):
         await repo.create_branch_in_git(branch_name=branch, branch_id=branch_id)
+
+
+@flow(name="artifact-definition-generate")
```
Comment (LucasG0) [inline backend/infrahub/git/tasks.py:114]: Currently `message_bus.operations.requests.artifact_definition.generate` has the same flow name. Just to make sure, is it what we want?

Comment (dgarros) [reply backend/infrahub/git/tasks.py:114]: Yes that's fine
Would be good to come up with a naming convention for the flow & deployment but that's outside of the scope of this PR

File: backend/tests/unit/message_bus/operations/event/test_branch.py [hunk 3]
```diff
@@ -84,21 +90,36 @@ async def test_merged(default_branch: Branch, init_service: InfrahubServices, pr
     mock_get_component_registry = MagicMock(return_value=mock_component_registry)
     mock_component_registry.get_component.return_value = diff_repo
 
-    with patch("infrahub.message_bus.operations.event.branch.get_component_registry", new=mock_get_component_registry):
+    with (
+        patch("infrahub.message_bus.operations.event.branch.get_component_registry", new=mock_get_component_registry),
+        patch(
+            "infrahub.services.adapters.workflow.local.WorkflowLocalExecution.submit_workflow"
+        ) as mock_submit_workflow,
+    ):
```
Comment (LucasG0) [inline backend/tests/unit/message_bus/operations/event/test_branch.py:98]: Before migrating to prefect, `generate` flow was not executed during the test as `test_merged` uses a `BusRecorder` and is only meant to  test that messages are correctly sent. Now using prefect workflows, `service.workflow.submit_workflow` actually executes `generate` flow, which raises an error due to an uninitialized client. So, patching `WorkflowLocalExecution.submit_workflow` fixes this issue, and allows to make sure flows have been correctly submitted.

File: backend/infrahub/git/tasks.py [hunk 4]
```diff
@@ -108,3 +109,16 @@ async def git_branch_create(
...
+async def generate(branch: str) -> None:
```
Comment (dgarros) [inline backend/infrahub/git/tasks.py:115]: I think it would be good to use a more descriptive name for this function like `artifact_definition_generate`

**Review by dgarros — APPROVED** [review]

File: backend/infrahub/git/tasks.py (hunk 4 above)
Comment (LucasG0) [reply backend/infrahub/git/tasks.py:115]: Right, I will modify it in a following PR migrating another task.

**Pull request merged**